tools/query_breakdown/src/main/java/com/google/bigquery/LocationTracker.java

```diff
@@ -0,0 +1,92 @@
+package com.google.bigquery;
+
+import java.util.ArrayList;
+
+/**
+ * This class tracks the original location of components in the query, thereby making sure that
+ * the error locations are correctly represented.
+ */
+public class LocationTracker {
```

Comment:
It's not clear to me how LocationTracker works. It seems that for each pair of (line, column), `location.get(line - 1).get(column-1)` will return the location. But I don't understand what the location value represents. It's an integer, so I guess it mean the index in the input string?

Reply:
Ah! So for each pair of (line, column), it is initialized in position at the (line - 1)th arraylist and (column - 1)th element of the arraylist as the integer column - 1. Since the line number won't ever change (but the column number will change constantly), we simply keep track of the original column number as the integer in the double arraylist

Reply:
Some question:
- Where is `location.get(x).get(y)` get assigned? 
- `ArrayList.add(x)` appends the value `x` to the array. It doesn't assign a value to `array[x]`. So in the method of  `public void add(int x, int y)`,  `location.get(x - 1).add(y)` doesn't seem right.

Reply:
yup! more documentation is now added